### ISSUE TITLE
Optimize plugin construction/run in packer phase 0

### ISSUE THREAD

**Issue by Pistonight** [Pistonite/celer#91]
Plugins may need extra power in the future like adding presets that need to be done before the comp phase.

Currently the compiler workflow is packer -> compiler -> plugin -> engine.

Could change plugin to be part of the packer and compiler phase:
packer -> packer (plugin) -> compiler -> compiler (plugin) -> engine

**Comment by Pistonight** [comment]
Repurposing this to track related performance issue.

Currently, both packer and compiler create `PluginRuntime`s to execute the plugin.

However, some plugins don't care about the packer phase 0 and can be skipped (i.e. no need to parse the settings and create the plugin).

I don't know if it's worth to optimize, since the result of packer phase 0 will be cached

**Comment by Pistonight** [comment]
No longer relevant after #172
